src/py/flwr/cli/new/templates/app/code/flwrtune/client.py.tpl

```diff
@@ -0,0 +1,127 @@
+"""$project_name: A Flower / flwrtune app."""
+
+from collections import OrderedDict
+from typing import Callable, Dict, Tuple
+from omegaconf import DictConfig
+import torch
+from trl import SFTTrainer
+from transformers import TrainingArguments
+from peft import get_peft_model_state_dict, set_peft_model_state_dict
+
+import flwr as fl
+from flwr.common.typing import NDArrays, Scalar
+
+from $import_name.models import get_model, cosine_annealing
+from $import_name.dataset import reformat
+
+
+# pylint: disable=too-many-arguments
+class FlowerClient(
+    fl.client.NumPyClient
+):  # pylint: disable=too-many-instance-attributes
+    """Standard Flower client for CNN training."""
+
+    def __init__(
+        self,
+        model_cfg: DictConfig,
+        train_cfg: DictConfig,
+        trainset,
+        tokenizer,
+        formatting_prompts_func,
+        data_collator,
+        save_path,
+    ):  # pylint: disable=too-many-arguments
+        self.device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        self.train_cfg = train_cfg
+        self.training_argumnets = TrainingArguments(**train_cfg.training_arguments)
+        self.tokenizer = tokenizer
+        self.formatting_prompts_func = formatting_prompts_func
+        self.data_collator = data_collator
+        self.save_path = save_path
+
+        # instantiate model
+        self.model = get_model(model_cfg)
+
+        self.trainset = trainset
+
+    def get_parameters(self, config: Dict[str, Scalar]) -> NDArrays:
+        """Return the parameters of the current net."""
+
+        state_dict = get_peft_model_state_dict(self.model)
+        return [val.cpu().numpy() for _, val in state_dict.items()]
```

Comment:
I'd suggest to use server-side parameter initialization and remove `get_parameters` altogether

Reply:
@danieljanes Thanks for the comments! All done~